src/components/ui/DataGrid/DataGrid.tsx

```diff
@@ -124,13 +136,32 @@ export function DataGrid<T>({
     };
   }, [error]);
 
+  const onFilterTextBoxChanged = (event: React.FormEvent<HTMLInputElement>) => {
+    if (gridRef.current?.api) {
+      gridRef.current.api.setGridOption(
+        "quickFilterText",
+        event.currentTarget.value
+      );
+    }
+  };
+
   return (
```

Comment:
Need to adjust the height of the table on this page so that adding the input doesn't cause an overflow.